Foundation/NSURLSession/NSURLSession.swift

```diff
@@ -258,7 +259,27 @@ open class URLSession : NSObject {
      * session with the same identifier until URLSession:didBecomeInvalidWithError: has
      * been issued.
      */
-    open func finishTasksAndInvalidate() { NSUnimplemented() }
+    open func finishTasksAndInvalidate() {
+       //we need to return immediately
+       workQueue.async {
+           //don't allow creation of new tasks from this point onwards
+           self.invalidated = true
```

Comment:
This seems a bit racy since invalidated is not checked on the same queue elsewhere, but since it's effectively a debugging tool I guess that's ok,